src/datepicker/datepicker.js

```diff
@@ -629,7 +629,17 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
         scope.$parent.$watch(getAttribute, function(value) {
           scope.watchData[key] = value;
           if (key === 'minDate' || key === 'maxDate') {
-            cache[key] = new Date(value);
+            var date = parseDate(value);
+            if (!date) {
+              date = new Date(value);
+              if (angular.isDate(date) && !isNaN(date)) {
+                var dateArray = date.toJSON().split('T')[0].split('-');
+                date = new Date(+dateArray[0], +dateArray[1] - 1, +dateArray[2]);
```

Comment:
That is why I mentioned earlier that it might be a good thing to make a breaking change and require that the user has passed in a date instead - if the user does `$scope.minDate = new Date(2001, 1, 2)`, and then uses the directive with `<uib-datepicker min-date="minDate" ng-model="date"></uib-datepicker>`, this would be an adequate fix without having to guess at timezones or put in hacks.